src/repl/scala/tools/nsc/interpreter/MemberHandlers.scala

```diff
@@ -35,7 +35,18 @@ trait MemberHandlers {
         // XXX this is obviously inadequate but it's going to require some effort
         // to get right.
         if (name.toString startsWith "x$") ()
-        else importVars += name
+        else {
+          importVars += name
+          if (name.endsWith(IMain.DummyCursorFragment)) {
+            val stripped = name.stripSuffix(IMain.DummyCursorFragment)
+            importVars += stripped
+            if (stripped.isTypeName)
+            // Needed to import `xxx` during line 2 of:
+            //   scala> val xxx = ""
+            //   scala> def foo: x<TAB>
+            importVars += stripped.toTermName
```

Comment:
The comment has to precede the if statement. I noticed it, too, but hoped the typesetter would catch it.